### DISCUSSION TITLE
Autoselect Search Box on page load?

### DISCUSSION THREAD

**Discussion by bryllyg** [microsoft-search/pnp-modern-search#1412] [Ideas]
**Version used**
4.3 and 3.21

**Question**
Is there a way to set the Search Box to autoselect on page load?  We have a home page with a search box on it and it seems like an additional step for users to then have to click in the search box before they start typing their query.

**Screenshots**

**Additional context**
I looked at the Search Box Properties, and don't see anything specific to autoselect.  Is this a no-go, or is there a super secret way ? :-)

Thanks!

**Comment by FranckyC** [comment]
Hi @bryllyg, actually what you describe as 'autoselect' is named 'focus' on a input HTML element. This behavior can be controlled in the SearchBox Office UI component but we we can't force it by default because we don't know about the page context and the way the search box is used. In your case it works but in another case, this can be annoying. I add it as an enhancement to add this option in the Web Part property.

**Comment by wobba** [comment]
And if multiple components ask for focus, it's not given which one "wins".

**Comment by bryllyg** [comment]
Yah, I was wondering about that. Would be nice to have the option available. thanks!

**Comment by bryllyg** [comment]
Follow up note, it would also be handy to have focus as a Web Part property on the Search Results web part. This would enable users to scroll through search results using the up and down arrows/page keys without first having to move the mouse and click on the results section.
